### PR TITLE
[ACCSEC-22227] - Manage biometric authentication errors.

### PR DESCRIPTION
**Pull request by AlejandroOrozco**
## Jira Ticket
ACCSEC-22227

## Github Issue
- ISSUE-

## Description
- Add BiometricError enum.
- Handler biometric errors using canEvaluatePolicy.

## Commit message
-  

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    case biometryNotAvailable
+    case biometryNotEnrolled
+
+    public var errorDescription: String? {
+        return "Biometric error: LAError \(String(describing: self))"
+    }
+
+    // MARK: - Resolution
+
+    public static func given(_ error: NSError?) -> Self? {
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift:34]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/84517577f916f06073de7ffb2ba09f81b7655c5d/TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift#L34"></span>Function should have complexity 10 or less: currently complexity equals 13<br />
<code>cyclomatic_complexity</code> <code>BiometricError.swift:34</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift [hunk 2]
```diff
@@ -36,12 +36,12 @@ public protocol AuthenticatedSecureStorageProvider {
...
-  private enum Errors: Error, LocalizedError {
-    case authenticationFailed
+  public enum Errors: Error, LocalizedError {
+    case authenticationFailed(Error?)
 
-    var errorDescription: String? {
+    public var errorDescription: String? {
       switch self {
-        case .authenticationFailed: return "Authentication failed"
+      case .authenticationFailed: return "Authentication failed"
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift:44]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/84517577f916f06073de7ffb2ba09f81b7655c5d/TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift#L44"></span>Case statements should be indented within their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>AuthenticatedSecureStorage.swift:44</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`95.65%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`94.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mejiagarcia — APPROVED** [review]
LGTM 👍

File: TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      guard let error = error else {
+        return nil
+      }
+
+      switch LAError.Code(rawValue: error.code) {
+        case .appCancel: return .appCancel
```
Comment (yafuquen) [inline TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift:40]: What is the difference between the cancels: app, user and system?

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift (hunk 2 above)
Comment (yafuquen) [reply TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift:44]: Could you check the indentation?

File: TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift (hunk 3 above)
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift:40]: The description is written in the LAError doc's. Maybe we can add just the link in the enums description https://developer.apple.com/documentation/localauthentication/laerror, so people can read the description from Apples doc? 
```swift
/// Authentication was canceled by application (e.g. invalidate was called while
/// authentication was in progress).
@available(iOS 9.0, *)
case appCancel = -9

// IE
authenticator.context.invalidate()

____________________________________

/// Authentication was canceled by user (e.g. tapped Cancel button).
public static var userCancel: LAError.Code { get }

____________________________________

/// Authentication was canceled by system (e.g. another application went to foreground).
public static var systemCancel: LAError.Code { get }

____________________________________
```

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift (hunk 2 above)
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift]: Fixed, Thank you 👍

File: TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift (hunk 1 above)
Comment (AlejandroOrozco) [reply TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift:36]: Removed complexity verification for Switch statement.

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.13%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`95.65%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`94.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by yafuquen — APPROVED** [review]

**Pull request merged**